pkg/webhook/webhook_test.go

```diff
@@ -85,51 +90,134 @@ func TestHandleWebhook(t *testing.T) {
+			expRespBody:             errNoPayload,
+			datastoreClientOverride: &FakeDatastoreClient{},
+		},
+		{
+			name:                    "invalid_signature",
+			pubSubGRPCConn:          pubSubGRPCConn,
+			dlqEventsPubSubGRPCConn: dlqEventsPubSubGRPCConn,
+			payloadFile:             path.Join(testDataBasePath, "pull_request.json"),
+			payloadType:             "pull_request",
+			payloadWebhookSecret:    "not-valid",
+			expStatusCode:           http.StatusUnauthorized,
+			expRespBody:             errInvalidSignature,
+			datastoreClientOverride: &FakeDatastoreClient{},
+		},
+		{
+			name:                    "error_write_backend_failed_bq_delivery_events_exists",
+			pubSubGRPCConn:          pubSubErrGRPCConn,
+			dlqEventsPubSubGRPCConn: dlqEventPubSubErrGRPCConn,
+			payloadFile:             path.Join(testDataBasePath, "pull_request.json"),
+			payloadType:             "pull_request",
+			payloadWebhookSecret:    serverGitHubWebhookSecret,
+			expStatusCode:           http.StatusInternalServerError,
+			expRespBody:             errWritingToBackend,
+			datastoreClientOverride: &FakeDatastoreClient{deliveryEventExists: &deliveryEventExistsRes{res: false, err: errors.New("error")}},
+		},
+		{
+			name:                    "idempotent_event_processing",
+			pubSubGRPCConn:          pubSubGRPCConn,
+			dlqEventsPubSubGRPCConn: dlqEventsPubSubGRPCConn,
+			payloadFile:             path.Join(testDataBasePath, "pull_request.json"),
+			payloadType:             "pull_request",
+			payloadWebhookSecret:    serverGitHubWebhookSecret,
+			expStatusCode:           http.StatusAlreadyReported,
+			expRespBody:             successMessage,
+			datastoreClientOverride: &FakeDatastoreClient{deliveryEventExists: &deliveryEventExistsRes{res: true}},
+		},
+		{
+			name:                    "error_write_backend_failed_marshal",
+			pubSubGRPCConn:          pubSubErrGRPCConn,
+			dlqEventsPubSubGRPCConn: dlqEventPubSubErrGRPCConn,
+			payloadFile:             path.Join(testDataBasePath, "pull_request.json"),
+			payloadType:             "pull_request",
+			payloadWebhookSecret:    serverGitHubWebhookSecret,
+			expStatusCode:           http.StatusInternalServerError,
+			expRespBody:             errWritingToBackend,
+			datastoreClientOverride: &FakeDatastoreClient{},
+		},
+		{
+			name:                    "error_write_backend_failed_bq_failure_events_exceeds_retry_limit",
+			pubSubGRPCConn:          pubSubErrGRPCConn,
+			dlqEventsPubSubGRPCConn: dlqEventsPubSubGRPCConn,
+			payloadFile:             path.Join(testDataBasePath, "pull_request.json"),
+			payloadType:             "pull_request",
+			payloadWebhookSecret:    serverGitHubWebhookSecret,
+			expStatusCode:           http.StatusInternalServerError,
+			expRespBody:             errWritingToBackend,
+			datastoreClientOverride: &FakeDatastoreClient{failureEventsExceedsRetryLimit: &failureEventsExceedsRetryLimitRes{res: false, err: errors.New("error")}},
+			// 152
```

Comment:
What are these numbers?

Reply:
slipped in, ill remove. They were to confirm line coverage in the unit test.